File: rsmdu/examples/rnb_from_api.rs
use anyhow::Result;
use rsmdu::geometric::rnb::Rnb;

/// Example: Loading RNB (French National Building Reference) data from RNB API
/// Following Python example from pymdu.geometric.Rnb
fn main() -> Result<()> {
    println!("=== Example: Loading RNB from RNB API ===\n");

    // Create Rnb instance
    // Python: rnb = Rnb(output_path='./')
    let mut rnb = Rnb::new(Some("./output".to_string()))?;

    // Set bounding box (La Rochelle, France)
    // Python: rnb.bbox = [-1.152704, 46.181627, -1.139893, 46.18699]
    rnb.set_bbox(-1.152704, 46.181627, -1.139893, 46.18699);

    println!("Bounding box set:");
    println!("  - Longitude: -1.152704 to -1.139893");
    println!("  - Latitude: 46.181627 to 46.18699");
    println!("  - Area: La Rochelle, France");
    println!("  - Format: WGS84 (EPSG:4326)\n");

    // Run RNB processing
    // Python: rnb = rnb.run()
    println!("Downloading and processing RNB from RNB API...");
    let rnb_result = rnb.run()?;

    println!("\nRNB processed successfully!");

    // Get GeoJSON (equivalent to to_gdf() in Python)
    if let Some(geojson) = rnb_result.get_geojson() {
        match geojson {
            geojson::GeoJson::FeatureCollection(fc) => {
                println!("  - Number of RNB buildings: {}", fc.features.len());
            }
            _ => {
                println!("  - GeoJSON loaded (non-FeatureCollection format)");
            }
        }
    }

    // Save to GeoJSON
    // Python: rnb.to_geojson(name="rnb")
    println!("\nSaving to GeoJSON...");
    rnb_result.to_geojson(None)?;

    println!("\n✅ Processing complete!");
    println!("  - Output file: {:?}", rnb_result.get_output_path());

    Ok(())
}

File: examples/rnb_from_api.py
"""
Example: Download RNB (French National Building Reference) data from RNB API using pymdurs

This example demonstrates how to:
1. Create a Rnb instance
2. Set a bounding box
3. Download RNB building data from RNB API
4. Get GeoJSON data
5. Save to GPKG file
"""

import pymdurs


def main():
    print("🏢 Loading RNB (French National Building Reference) from RNB API...")

    # Create Rnb instance
    rnb = pymdurs.geometric.Rnb(output_path="./output")

    # Set bounding box (La Rochelle area, France)
    # Format: min_x, min_y, max_x, max_y (WGS84, EPSG:4326)
    rnb.set_bbox(-1.152704, 46.181627, -1.139893, 46.18699)

    # Set CRS (optional, defaults to EPSG:2154)
    rnb.set_crs(2154)

    geo = rnb.geo_core
    print(f"📦 Bounding box set")
    print(f"🗺️  CRS: {geo.epsg}")
    print(f"📁 Output path: {geo.output_path}")

    # Run RNB processing: fetches from RNB API and parses JSON
    print("⏳ Downloading RNB data from RNB API...")
    rnb = rnb.run()

    # Get GeoJSON (equivalent to to_gdf() in Python)
    print("📊 Getting GeoJSON data...")
    geojson = rnb.get_geojson()

    if geojson and "features" in geojson:
        num_features = len(geojson["features"])
        print(f"✅ Loaded {num_features} RNB buildings")
    else:
        print("✅ RNB data loaded")

    # Save to GPKG
    print("💾 Saving to GPKG...")
    rnb.to_geojson(name="rnb")

    print(f"✅ RNB processing complete!")
    print(f"📁 Output path: {rnb.get_output_path()}")

    return rnb


if __name__ == "__main__":
    rnb = main()
